src/providers/LyricSify.ts

```diff
@@ -0,0 +1,71 @@
+import { fetchText } from '../utils'
+import { Provider, SearchParams } from './Provider'
+
+const BASE_URL = 'https://www.lyricsify.com/'
+
+export class LyricSify implements Provider {
+    private async getLink(artist: string, name: string) {
+        const query = artist + ' ' + name;
+        const data = await fetchText(BASE_URL + 'search?q=' + query);
+        const result = data.toString();
+        const list = result.split('<div class="li">');
+        const parsed = list.slice(1, list.length);
+        var links;
+        parsed.forEach((items) => {
+            const title = items.match('>(.*?)<')
+            const link = items.match('href=\"([^"]*)\"')
+            if (title?.length && link?.length) {
+                if (title[1].toLowerCase().includes(artist) && title[1].toLowerCase().includes(name)) {
+                    links = link[1];
+                }
+            }
+        });
+
+        return links;
+    }
+
+    private removeTags = (str: string) => {
+        if ((str === null) || (str === ''))
+            return false;
+        else
+            str = str.toString().replace(/(<([^>]+)>)/ig, '')
+                .replace(/&lt;/g, "<")
+                .replace(/&gt/g, ">")
+                .replace(/&quot;/g, "\"")
+                .replace(/^\s*\n\n/gm, '')
+                .replace(/&#039;/g, "'")
+                .replace(/&amp;/g, "&");
+        const listStr = str.split("\n");
+        var array: Array<string>;
+        array = [];
+        listStr.forEach((items) => {
+            if (items.substring(0, 9).includes('.')) {
+                array.push(items);
+            }
+        })
```

Comment:
Should we keep the meta lines, for example the artist, title and offset?

I noticed that there is a Download button on the page. Instead of parsing the raw HTML, maybe we can just follow the download link.